### PR TITLE
Submitting for review

### PR DESCRIPTION
**Pull request by motanelu**
I know that it's outside the "competition", but the technology stack is really cool and I'd appreciate your feedback. Thank you.

### CODE REVIEW

**Comment by mwpicard** [comment]
Hello Tudor

cool indeed. Mauro and I are discussing this tonight and will respond
then.  I really appreciate the time you put in.

Where are you living these days?

We should chat on skype soon.

cheers

Martin

On Sunday, February 22, 2015, Tudor Barbu <notifications@github.com> wrote:

> I know that it's outside the "competition", but the technology stack is
> really cool and I'd appreciate your feedback. Thank you.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Vreasy/VrRecruit/pull/14
> Commit Summary
>
>    - Interpreter class & test
>    - New models
>    - Display properly the errors with the default layout (for debug
>    reasons)
>    - Callback controller - handle incoming Twilio requests
>    - Tests for incoming Twilio requests
>    - Updated db dump for testing
>    - Frontend application - new REST request and angular implementation
>    - New migration files
>
> File Changes
>
>    - *A* library/Vreasy/Models/History.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-0> (133)
>    - *M* library/Vreasy/Models/Task.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-1> (66)
>    - *A* library/Vreasy/Utils/Interpreter.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-2> (141)
>    - *M* migrations/webapp/20140120145548_InsertSomeTasks.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-3> (2)
>    - *A* migrations/webapp/20150222111312_HistoryAndStatusesForTasks.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-4> (39)
>    - *A* migrations/webapp/20150222112621_InsertSomeHistoryData.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-5> (34)
>    - *M* tests/_data/dump.sql
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-6> (13)
>    - *A* tests/functional/CallbackFailCept.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-7> (44)
>    - *A* tests/functional/CallbackSuccessCept.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-8> (43)
>    - *A* tests/unit/InterpreterTest.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-9> (91)
>    - *M* vreasy/application/configs/application.ini
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-10> (1)
>    - *M* vreasy/application/layouts/scripts/layout.phtml
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-11> (20)
>    - *A* vreasy/application/layouts/scripts/xml.phtml
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-12> (10)
>    - *A*
>    vreasy/application/modules/default/controllers/CallbackController.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-13> (157)
>    - *M*
>    vreasy/application/modules/default/controllers/ErrorController.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-14> (4)
>    - *A*
>    vreasy/application/modules/default/views/scripts/callback/message-received.phtml
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-15> (10)
>    - *M* vreasy/application/modules/vreasy/controllers/TaskController.php
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-16> (1)
>    - *M* vreasy/public/taskConfirmationApp/app.js
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-17> (30)
>    - *M* vreasy/public/taskConfirmationApp/templates/index.html
>    <https://github.com/Vreasy/VrRecruit/pull/14/files#diff-18> (28)
>
> Patch Links:
>
>    - https://github.com/Vreasy/VrRecruit/pull/14.patch
>    - https://github.com/Vreasy/VrRecruit/pull/14.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Vreasy/VrRecruit/pull/14>.
>


-- 
Martin Picard <http://www.linkedin.com/in/martinpicard>
Founder
<http://www.vreasy.com>
Vreasy Corporation <http://www.vreasy.com>
*HQ:*
7800 N MoPac,
Austin, TX 78759
*R&D  - MAP <http://bit.ly/VreasyMAP>*
57 Atlantida
Barcelona 08003
Tel: +34606551138
Email: martin@vreasy.com
Skype: skyper
<http://bit.ly/1pbcdfy>

*Vreasy.com* wants YOU for our Barcelona R&D offices. Join our talented
international team and work minutes from the beach.  Be inspired to
achieve.  Check out our Jobs Board <http://bit.ly/1pbcdfy>

**Comment by motanelu** [comment]
Hi Martin, just saw your comment now. Sure thing, you guys have an awesome technology stack. 

t.

File: vreasy/public/taskConfirmationApp/app.js [hunk 1]
```diff
@@ -24,6 +34,22 @@ angular.module('taskConfirmationApp',  ['ui.router', 'ngResource'])
...
 })
-.controller('TaskCtrl', function($scope, Task) {
+.controller('TaskCtrl', function($scope, $http, Task) {
     $scope.tasks = Task.query();
+    $scope.active = {};
+    $scope.custom = false;
+
+    $scope.display = function(id) {
+        if ($scope.active.id != id) {
+            $http.get('/task/' + id + '?format=json')
```
Comment (brutuscat) [inline vreasy/public/taskConfirmationApp/app.js]: @motanelu any reason why u haven't used ngResource factory declared above?

Comment (motanelu) [inline vreasy/public/taskConfirmationApp/app.js]: I'm just getting started with AngularJS and I'm not yet familiar with its best practices.

Comment (brutuscat) [inline vreasy/public/taskConfirmationApp/app.js]: @motanelu no worries then!

**Pull request closed without merging**